### PR TITLE
WIP: Add jcvi.graphics.ribbon module

### LINKED ISSUE #724: Support custom alignment blocks in Microsynteny plots
Implement `jcvi.graphics.ribbon` module as a fork of `synteny` that supports plotting of arbitrary links between chromosomes. 

The `synteny` module currently draws links between genes that are defined as orthologues. Several users have asked for a feature that essentially plots pairwise chromosome alignments (i.e. from BLAT, LASTZ, Minimap2 etc) instead of just linking syntenic genes.

- [x] Add ribbon module
- [ ] Add tool to run LASTZ and format output for ribbon

### PR DESCRIPTION
**Pull request by Adamtaranto**
When complete, this PR will add a new module `jcvi.graphic.ribbon` which is derived from the synteny module. 

Closes #724

Ribbon adds support for plotting whole genome alignments instead of linked orthologous genes.

- [ ] Set contig orientation
- [ ] Set contig zoom interval
- [ ] Manually set ribbon colours
- [ ] Convert LASTZ to ribbon link format
- [ ] Set annotation track offsets
- [ ] Set annotation track colours
- [ ] Add tests for processing new config format
- [ ] Use base logger

### CODE REVIEW

File: jcvi/graphics/base.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 #!/usr/bin/env python
 # -*- coding: UTF-8 -*-
 
+from os import remove
 import copy
+import logging
```
Comment (tanghaibao) [inline jcvi/graphics/base.py:6]: nit: there is a package-wide `logger` that you can use.

```python
from ..apps.base import logger
```

Then use `logger.debug()` below.

File: jcvi/graphics/ribbon.py [hunk 2]
```diff
@@ -0,0 +1,886 @@
...
+0.5, 0.4, 0, right, center, m, 1, Spp 2 Chrm 1.4 , Scaffold_1.4, 50000, 200000, 2787645
+0.5, 0.2, 45, bottom, center, b, 1, Spp3 Chr 5, SppX_Chromosome_05, 60000, 460000, 2046703
+# edges
+e, 0, 1
+e, 1, 2
+
+With the row ordering corresponding to the column ordering in the alignment blockfile.
+"""
+
+import logging
```
Comment (tanghaibao) [inline jcvi/graphics/ribbon.py:21]: Same comment as above.

File: temp/ribbon_testing/data/3S_genes.bed [hunk 3]
```diff
@@ -0,0 +1,2239 @@
+SN15_SOL_Chromosome_05	4558	5204	.	.	-
```
Comment (tanghaibao) [inline temp/ribbon_testing/data/3S_genes.bed:1]: Why in `temp` directory? Perhaps just move these `ribbon_testing` to regular testing if they pass?

**Review by tanghaibao — COMMENTED** [review]
Looks great and thanks for adding this module. Some minor comments then we should be able to merge.

File: temp/ribbon_testing/data/3S_genes.bed (hunk 3 above)
Comment (Adamtaranto) [reply temp/ribbon_testing/data/3S_genes.bed:1]: oops, local test. Shouldn't be here.

File: jcvi/graphics/ribbon.py (hunk 2 above)
Comment (Adamtaranto) [reply jcvi/graphics/ribbon.py:21]: Switch to apps.base.logger

**Comment by Adamtaranto** [comment]
Note: Black formatter action seems to fail when PR is not owned by tanghaibao/jcvi. 

Need to look into this.

**Review by tanghaibao — APPROVED** [review]

**Pull request merged**